src/io/flutter/sdk/FlutterSdk.java

```diff
@@ -61,6 +70,59 @@ public static FlutterSdk getFlutterSdk(@NotNull final Project project) {
     return project.isDisposed() ? null : getFlutterSdkByDartSdk(DartPlugin.getDartSdk(project));
   }
 
+
+  /**
+   * Returns the Flutter SDK for a project that has a possibly broken "Dart SDK" project library.
+   * <p>
+   * (This can happen for a newly-cloned Flutter SDK where the Dart SDK is not cached yet.)
+   */
+  @Nullable
+  public static FlutterSdk getIncomplete(@NotNull final Project project) {
```

Comment:
So, this method will always return a FlutterSdk if one is set, whether it's fully set up or not? We may want to rethink the names for this and the current `getFlutterSdk()` to clarify that this method always returns an sdk, and the other only returns fully formed sdks.